src/map/skill.cpp

```diff
@@ -7505,14 +7501,13 @@ int32 skill_castend_damage_id (struct block_list* src, struct block_list *bl, ui
 		clif_skill_damage( *src, *bl, tick, status_get_amotion(src), tstatus->dmotion,
 			0, abs(skill_get_num(skill_id, skill_lv)),
 			skill_id, skill_lv, skill_get_hit(skill_id) );
-		map_freeblock_unlock();
 		return 1;
 	}
 
 	if( sc && sc->getSCE(SC_CURSEDCIRCLE_ATKER) ) //Should only remove after the skill has been casted.
 		status_change_end(src,SC_CURSEDCIRCLE_ATKER);
 
-	map_freeblock_unlock();
+	freeLock.unlock();
```

Comment:
Unrelated to this PR, but isnt this too early here?
"sd" is still used below here and may be freed by calling unlock here.
(See also skill_castend_nodamage_id, there it is called at the correct location, or so to say "not called" now anymore, since we use the scope destructor now).

Reply:
I think that would be up to the next PR to see if we need to move it. I'd like to keep it here to lower the amount of changes.